### DISCUSSION TITLE
How to interrupt a form submission

### DISCUSSION THREAD

**Discussion by xiaomlove** [filamentphp/filament#3185] [Help]
For example, if I customize the creation process when creating a new one, and an exception is thrown in the middle, how do I interrupt and return the exception message to frontend? I know I can override the whole create() method and do notify(), but is there any way to do this in handleRecordCreation().

```
protected function handleRecordCreation(array $data): Model
{
    $seedBoxRep = new SeedBoxRepository();
    try {
        return $seedBoxRep->store($data);
    } catch (\Exception $exception) {
        //this wont work...
        $this->notify('danger', $exception->getMessage());
        die();
    }
}
```

**Comment by xiaomlove** [comment]
Not only the creation process, but I think any [Lifecycle hook](https://filamentphp.com/docs/2.x/admin/resources/creating-records#lifecycle-hooks) should be able to abort the whole request in advance.

**Comment by danharrin** [comment]
Surely just implementing good validation will solve this issue? You shouldnt be reporting exceptions as notifications to the user, that is really really bad security practice. If something goes wrong in the save process then it shouldnt have been because of user error, the app is broken. User error is dealt with by validation.
